dali/imgcodec/parsers/png.cc

```diff
@@ -12,19 +12,99 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#include "dali/core/byte_io.h"
 #include "dali/imgcodec/parsers/png.h"
 
 namespace dali {
 namespace imgcodec {
 
+// https://www.w3.org/TR/2003/REC-PNG-20031110
+
+enum ColorType : uint8_t {
+  PNG_COLOR_TYPE_GRAY       = 0,
+  PNG_COLOR_TYPE_RGB        = 2,
+  PNG_COLOR_TYPE_PALETTE    = 3,
+  PNG_COLOR_TYPE_GRAY_ALPHA = 4,
+  PNG_COLOR_TYPE_RGBA       = 6
+};
+
+struct IhdrChunk {
+  uint32_t width;
+  uint32_t height;
+  uint8_t color_type;
+  // Some fields were ommited.
+
+  int GetNumberOfChannels() {
+    switch (color_type) {
+      case PNG_COLOR_TYPE_GRAY:
+      case PNG_COLOR_TYPE_GRAY_ALPHA:
+        return 1;
```

Comment:
```suggestion
      case PNG_COLOR_TYPE_GRAY:
        return 1;
      case PNG_COLOR_TYPE_GRAY_ALPHA:
        return 2;
```
According to the page you quoted:
PNG image type | Colour type | Allowed bit depths |  
-- | -- | -- | --
Greyscale with alpha | 4 | 8, 16 | Each pixel is a greyscale sample followed by an alpha sample.

Reply:
This bit of code was taken from our old parser, I assumed it was correct :/

Reply:
Well, it wasn't.